frontend/ui/widget/keyvaluepage.lua

```diff
@@ -543,6 +544,90 @@ function KeyValuePage:_populateItems()
+    local key_w = math.floor(frame_internal_width / 2 - middle_padding)
+    local value_w = math.floor(frame_internal_width / 2)
+
+    local key_widget = TextWidget:new{
+        text = " ",
+        max_width = available_width,
+        face = Font:getFace("smallinfofontbold", self.items_font_size),
+    }
+    local value_widget = TextWidget:new{
+        text = " ",
+        max_width = available_width,
+        face = Font:getFace("smallinfofont", self.items_font_size),
+        lang = self.values_lang,
+    }
+    local key_widths = {}
+    local value_widths = {}
+    local tvalue
+    for idx=1, self.items_per_page do
+        local kv_pairs_idx = idx_offset + idx
+        local entry = self.kv_pairs[kv_pairs_idx]
+        if entry == nil then break end
+        if type(entry) == "table" then
+            tvalue = tostring(entry[2])
+            tvalue = tvalue:gsub("[\n\t]", "|")
+
+            key_widget:setText(entry[1])
+            value_widget:setText(tvalue)
+
+            table.insert(key_widths, key_widget:getWidth())
+            table.insert(value_widths, value_widget:getWidth())
+        end
+    end
+
+    table.sort(key_widths)
+    table.sort(value_widths)
+
+    local misplace_count -- count item that needs to displace or truncate key/value
+    -- first we check if no displacement needed at all
+    local width_ratio
+    if key_widths[#key_widths] <= key_w and value_widths[#value_widths] <= value_w then
+        width_ratio = 1/2
+    end
+    if not width_ratio then
+        -- has to displace
+        for vi = #value_widths, 1, -1 do
+            local key_width_limit = available_width - value_widths[vi]
+            local key_cut_count = 0 -- if we were to draw a line at the beginning of this value, how many keys'll be cut
+            for ki = #key_widths, 1, -1 do
+                if key_widths[ki] > key_width_limit then
+                    key_cut_count = key_cut_count + 1
+                else
+                    break -- others are all shorter so no more cut
+                end
+            end
+            local total_cut_count = key_cut_count + (#value_widths - vi) -- latter is value_cut_count
+
+            if misplace_count then -- not the first round of iteration
+                if total_cut_count > misplace_count then
+                    -- previous iteration has the least moved ones
+                    width_ratio = (available_width - value_widths[vi+1] + middle_padding) / frame_internal_width
+                    break
+                elseif total_cut_count == misplace_count then
+                    -- both have the least cut ones, we take the mean
+                    width_ratio = (available_width - (value_widths[vi+1] + value_widths[vi])/2 + middle_padding) / frame_internal_width
+                    break
+                else
+                    -- still could be less
+                    misplace_count = total_cut_count
```

Comment:
I'm having a hard time to get why to total cut count becomes a misplace (or displace) count,. This is probably logical, but some more comments so I (and others) don't have to scratch my head would be nice :)
(Also, displace or misplace sounds odd (déplacement in French, so the meaning is clear to me), but with dis/mis if feels something is wrong, wrongly placed - probably even more odd with "mis"place.)

Reply:
Changed the wording to `unfit`, should be better. Et je sais un peu français, peut-être ça m'a fait choisir le mot :)
And added more comment to explain the logic, should be clearer now.

Reply:
Your french is nice ! Mais il faudrait travailler un peu ton anglais maintenant :)

Reply:
Oui. We don't have plural, tense nor conjugation. The habits often slip in.